src/authorization/signin.js

```diff
@@ -0,0 +1,273 @@
+import Axios from "axios";
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { Link } from 'react-router-dom';
+import { Redirect } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import actionCreators from '../states/index';
+import bcrypt from 'bcryptjs'
+function Signin() {
+  //used to send action to reducer
+  const dispatch = useDispatch();
+  //will creator fuction to send action to reducer
+  const { signin } = bindActionCreators(actionCreators, dispatch)
+  //will downlink data from store.session in const
+  const loggedIn = useSelector(state => state.session)
+  const [signinDetails, setSigninDetails] = useState({
+    "email": '',
+    "password": ''
+  })
+  const submitAsAdmin = async () => {
+    console.log("submit as admin")
+    var item = {
+      "admin_email": signinDetails.email,
+      //"admin_password": signinDetails.password
+    }
+    console.log(JSON.stringify(item))
+    Axios.post('/api/signin/admin', item, { headers: { 'content-type': 'application/json' } })
+      .then(
+        async response => {
+          console.log(response.data)
+          const salt = response.data.salt
+          const EncryptedPass = await bcrypt.hash(signinDetails.password, salt)
+          if (EncryptedPass === response.data.admin_password) {
+            toast.success("logged in with Admin_id " + response.data.admin_id + " as " + response.data.admin_type, {
+              position: "top-center",
+              autoClose: 5000,
+              hideProgressBar: false,
+              closeOnClick: true,
+              pauseOnHover: true,
+              draggable: true,
+              progress: undefined,
+            });
+            const whoIsLogged = {
+              "id": response.data?.admin_id,
+              "name": response.data?.admin_name,
+              "email": signinDetails?.email,
+              "password": response.data?.admin_password
+            }
+            signin(whoIsLogged)
+          }
+          else {
+            toast.error("Invalid Password",
+              {
+                position: "top-center",
+                autoClose: 5000,
+                hideProgressBar: false,
+                closeOnClick: true,
+                pauseOnHover: true,
+                draggable: true,
+                progress: undefined,
+              });
+          }
+
+        }
+
+      )
+      .catch(
+        error => {
+          console.log(error.response)
+          toast.error("Some thing went wrong \n " + JSON.stringify(error.response.data),
+            {
+              position: "top-center",
+              autoClose: 5000,
+              hideProgressBar: false,
+              closeOnClick: true,
+              pauseOnHover: true,
+              draggable: true,
+              progress: undefined,
+            });
+
+        });
+  }
+  const submitAsUser = async () => {

```

Comment:
why are we having redundant code one for `user` and `admin`. why do we need this? we could use a function that reduces redundancy